### ISSUE TITLE
3.0 iOS Hass App opens fullscreen stream

### ISSUE THREAD

**Issue by macchrissli** [AlexxIT/WebRTC#388]
Since i updated the dashboard on the Hass iOS app always opens the webrtc stream to fullscreen mode. Makes it unusable to have as a card at the moment

**Comment by andker87** [comment]
same here, full-screen display automatically occurs even if you are not on the page where the cam previews are. I had to rollback to 2.3.1 because the home assistant app was unusable

**Comment by AlexxIT** [comment]
What iOS version do you use? Is WebRTC card on default page?

**Comment by andker87** [comment]
iOS 16.2 on iPhone X.
WebRTC is in a tab on the default dashboard, but not on the main page.

**Comment by PablaV** [comment]
I am also experiencing this on the companion app. Running iOS 16.2 on a iPhone 14 pro max.

**Comment by Yamaha0014** [comment]
I have the same problem, Iphone 13 pro and ios 16.2 last version WebRTC and go2rtc

**Comment by jason0x43** [comment]
I am also seeing this issue on an iPhone 13 Pro running iOS 16.2, Companion app 2022.10.1, and WebRTC 3.0.0. I have a dashboard page with 3 custom:webrtc-camera cards:

```yaml
cards:
  - type: custom:webrtc-camera
    entity: camera.cat_room
    muted: true
    ui: false
  - type: custom:webrtc-camera
    entity: camera.back_yard
    muted: true
    ui: false
  - type: custom:webrtc-camera
    entity: camera.garage
    muted: true
    ui: false
```

When the dashboard loads, the three camera cards start loading their streams. As each stream loads, it becomes full screen. I also have dashboard pages for each individual camera. The same thing happens on those -- as soon as the camera stream loads, the video becomes full screen.

This isn't 100% consistent. The video becoming full screen happens most of the time when I open a page with a webrtc-camera card, but not every time.

**Comment by jason0x43** [comment]
For me, at least, this behavior only seems to happen in the iPhone Companion app. The Companion app on an iPad, and Safari on any device, behave normally.

**Comment by dm82m** [comment]
I have a weird behavior aswell: iPhone with companion app and local network: all good. Fluent video, … If I disable wifi and navigate to the tab with webrtc it opens it in fullscreen and also seems like loading 3 frames playing them, loading three frames and playing them. Mobile connection is quite good, so no buffering issue. Updated from 2.x version and had the Udp 5000 ports forwarded and replaced that with tcp/udp 8555 and added stun:8555 to configuration.yaml aswell.

**Comment by dm82m** [comment]
I have the same issue on iPhone with companion but only on cellular connection if the stream is chunking. On local network the issue does not appear!

**Comment by Subtielbeek** [comment]
Same issue here iPhone 13 pro max, iOS 16.0, Home Assistant 2023.1.2, Companion App 2022.10.1 (2022.424)

**Comment by AlexxIT** [comment]
As temporary solution you can disable `mp4` mode:
```
mode: webrtc,mse,mjpeg
```

But none of you will see the stream from the camera at all. Because:
- webrtc won't work for you because you don't setup external access
- mse won't work for you because you have iPhone...
- mp4 won't work for you because full screen problem (will try to fix it)

**Comment by jason0x43** [comment]
Setting mode to `webrtc,mse` resolves the issue in the companion app on my iPhone — videos start playing normally, without automatically switching to full screen.

**Comment by Subtielbeek** [comment]
> mode: webrtc,mse,mjpeg

Can confirm that this works! 😄

**Comment by bm55** [comment]
Yes removing mp4 does fixes the iPhone issue

**Comment by AlexxIT** [comment]
Fixed in v3.0.1
